src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs

```diff
@@ -47,6 +47,11 @@ public static bool IsLambda(SyntaxNode node)
         public static SyntaxNode GetLambda(SyntaxNode lambdaBody)
         {
             var lambda = lambdaBody.Parent;
+            if (lambda is ArrowExpressionClauseSyntax)
```

Comment:
>lambda [](start = 16, length = 6)

Use .Kind() == SyntaxKind.ArrowExpression instead.

Is this addressing local function with expression body? If so, please add a comment and an assert that the lambda.Parent is a local function statement syntax